packages/react-native-renderer/src/ReactFabricRenderer.js

```diff
@@ -0,0 +1,321 @@
+      rootContainerInstance, // rootTag
+      {text: text}, // props
+      internalInstanceHandle, // instance handle
+    );
+
+    return {
+      node: node,
+    };
+  },
+
+  finalizeInitialChildren(
+    parentInstance: Instance,
+    type: string,
+    props: Props,
+    rootContainerInstance: Container,
+  ): boolean {
+    return false;
+  },
+
+  getRootHostContext(): {} {
+    return emptyObject;
+  },
+
+  getChildHostContext(): {} {
+    return emptyObject;
+  },
+
+  getPublicInstance(instance) {
+    return instance.canonical;
+  },
+
+  now: ReactNativeFrameScheduling.now,
+
+  prepareForCommit(): void {
+    // Noop
+  },
+
+  prepareUpdate(
+    instance: Instance,
+    type: string,
+    oldProps: Props,
+    newProps: Props,
+    rootContainerInstance: Container,
+    hostContext: {},
+  ): null | Object {
+    const viewConfig = instance.canonical.viewConfig;
+    const updatePayload = ReactNativeAttributePayload.diff(
+      oldProps,
+      newProps,
+      viewConfig.validAttributes,
+    );
+    // TODO: If the event handlers have changed, we need to update the current props
+    // in the commit phase but there is no host config hook to do it yet.
+    return updatePayload;
+  },
+
+  resetAfterCommit(): void {
+    // Noop
+  },
+
+  scheduleDeferredCallback: ReactNativeFrameScheduling.scheduleDeferredCallback,
+  cancelDeferredCallback: ReactNativeFrameScheduling.cancelDeferredCallback,
+
+  shouldDeprioritizeSubtree(type: string, props: Props): boolean {
+    return false;
+  },
+
+  shouldSetTextContent(type: string, props: Props): boolean {
+    // TODO (bvaughn) Revisit this decision.
+    // Always returning false simplifies the createInstance() implementation,
+    // But creates an additional child Fiber for raw text children.
+    // No additional native views are created though.
+    // It's not clear to me which is better so I'm deferring for now.
+    // More context @ github.com/facebook/react/pull/8560#discussion_r92111303
+    return false;
+  },
+
+  persistence: {
+    cloneInstance(
+      instance: Instance,
+      updatePayload: null | Object,
+      type: string,
+      oldProps: Props,
+      newProps: Props,
+      internalInstanceHandle: Object,
+      keepChildren: boolean,
+      recyclableInstance: null | Instance,
+    ): Instance {
+      const node = instance.node;
+      let clone;
+      if (keepChildren) {
+        if (updatePayload !== null) {
+          clone = FabricUIManager.cloneNodeWithNewProps(node, updatePayload);
+        } else {
+          clone = FabricUIManager.cloneNode(node);
+        }
+      } else {
+        if (updatePayload !== null) {
+          clone = FabricUIManager.cloneNodeWithNewChildrenAndProps(
+            node,
+            updatePayload,
+          );
+        } else {
+          clone = FabricUIManager.cloneNodeWithNewChildren(node);
+        }
+      }
+      return {
+        node: clone,
+        canonical: instance.canonical,
```

Comment:
This is intentionally shared between all clones, right? Is cloning just an implementation detail invisible to the user, like with Fibers?